### PR TITLE
Prevent a panic in the error encoder

### PR DESCRIPTION
**Pull request by richard-rance**
The Parameter can be nil when the enum is set of one of the body fields.

### CODE REVIEW

**Comment by fenollp** [comment]
Thanks for catching this & going all the way to a patch.

Could you provide a test that would fail if it weren't for your PR, please?

**Comment by richard-rance** [comment]
I've added couple of tests and fixed another error I noticed.

File: openapi3filter/validation_error_encoder.go [hunk 1]
```diff
@@ -148,7 +148,7 @@ func convertSchemaError(e *RequestError, innerErr *openapi3.SchemaError) *Valida
 	}
 
 	// Add error source
-	if e.Parameter != nil && e.Parameter.In == "query" {
```
Comment (fenollp) [inline openapi3filter/validation_error_encoder.go:151]: Why is this needed?

Comment (richard-rance) [reply openapi3filter/validation_error_encoder.go:151]: This was the other issue I found. There were no tests for parameters in other locations. When I added a test for one in the header the source was not getting set.
